Slot Machine false positives

ahstro opened the pull request:
The 'Add your JavaScript Slot Machine Slots' waypoint would throw false positives if all slot numbers were the same, effectively allowing the campers to pass the test without having written any code.

This would fix it, but since it alters the challenge seed, the seed for this challenge would look different from close neighbours, so they might need to be altered as well. I'll look through the other challenges and see if they can be changed, but I wanted to get this up ad quickly as possible to hear what others have to say.

Fixes #3228

ahstro commented:
I'm a bit confused regarding what is expected from the camper. The instructions say `If all three numbers match, we should return the number that we have in three of slots or leave it as null.`, but the only addition to the seed for the next challenge is
```javascript
if(slotOne !== slotTwo || slotTwo !== slotThree){
  return null;
}
```
From the wording in the instructions, I would expect the solution to look more like this
```javascript
if(slotOne === slotTwo && slotTwo === slotThree){
  return slotOne;
} else {
  return null;
}
```
If we expected that, then the test could be
```javascript
assert((function(){
  var data = runSlots();
  return data.length === 1 || data === null;
})()
```
and we wouldn't have to change the seed, but it feels a little bit strange. I'm going to need someone else's opinion on this one.

BerkeleyTrue commented:
indeed it should be:
```
if(slotOne === slotTwo && slotTwo === slotThree){
  return slotOne;
}
// avoid else and use `return early pattern`
return null;
```

ahstro commented:
I've updated the test to make sense if that code is what we're expecting.

I just realized that I forgot to update the seed for the next challenge; should I just replace
```javascript
if(slotOne !== slotTwo || slotTwo !== slotThree){
  return null;
}
```
with 
```javascript
if(slotOne === slotTwo && slotTwo === slotThree){
  return slotOne;
}
return null;
```
?

BerkeleyTrue commented:
Yes this looks great:
```js
if(slotOne === slotTwo && slotTwo === slotThree){
  return slotOne;
}
return null;
```

Commit:
Slot Machine false positives

The 'Add your JavaScript Slot Machine Slots' waypoint would
throw false positives if all slot numbers were the same, effectively
allowing the campers to pass the test without having written any code.

ahstro commented:
Blargh.. There's something wrong with my brain today. This should do it, I think.

BerkeleyTrue commented:
@ahstro What is the status of this PR?

ahstro commented:
@BerkeleyTrue It should be finished now, but I'd love it if you could QA it quickly.
